src/key_io.cpp

```diff
@@ -83,14 +83,22 @@ class DestinationEncoder
 
 CTxDestination DecodeDestination(const std::string& str, const CChainParams& params, std::string& error_str, std::vector<int>* error_locations)
 {
-    std::vector<unsigned char> data;
-    uint160 hash;
     error_str = "";
 
-    // Note this will be false if it is a valid Bech32 address for a different network
-    bool is_bech32 = (ToLower(str.substr(0, params.Bech32HRP().size())) == params.Bech32HRP());
+    static const uint8_t MAX_BASE58_CHARS = 100;
+    static const uint8_t MAX_BASE58_CHECK_CHARS = 21;
+
+    std::vector<unsigned char> data, bech32_data;
+
+    auto [bech32_encoding, bech32_hrp, bech32_chars] = bech32::Decode(str);
+    auto [bech32_error, bech32_error_loc] = bech32::LocateErrors(str);
+    bool is_bech32 = bech32_encoding != bech32::Encoding::INVALID;
```

Comment:
does it even make sense to run `bech32::LocateErrors` when `bech32_encoding != bech32::Encoding::INVALID`? Seems a bit all over the place.